internal/security/proxy/service.go

```diff
@@ -171,13 +175,18 @@ func (s *Service) postCert(cert CertificateLoader) error {
 	default:
 		b, err := ioutil.ReadAll(resp.Body)
 		if err != nil {
-			return err
+			return err, false
 		}
-		e := fmt.Sprintf("failed to add certificate with errorcode %d, error %s", resp.StatusCode, string(b))
+		reason := string(b)
+		e := fmt.Sprintf("failed to add certificate with errorcode %d, error %s", resp.StatusCode, reason)
 		LoggingClient.Error(e)
-		return errors.New(e)
+		if (resp.StatusCode == http.StatusBadRequest) && (strings.Index(reason, "existing certificate") != -1) {
+			e = fmt.Sprintf("certificate already exists on reverse proxy")
+			return nil, true
+		}
+		return errors.New(e), false
 	}
-	return nil
+	return nil, false
```

Comment:
should return `true` here, too.
```suggestion
	return nil, true
```

Reply:
This boolean means "existed" not "was successful"